lib/winapi/timezoneapi.c

```diff
@@ -0,0 +1,196 @@
+    TIME_FIELDS dateNow;
+    LARGE_INTEGER kTime;
+    ULONG type;
+
+    RtlZeroMemory(lpTimeZoneInformation, sizeof(TIME_ZONE_INFORMATION));
+
+    queryStatus = ExQueryNonVolatileSetting(XC_TIMEZONE_BIAS, &type, &timeZoneBias, sizeof(timeZoneBias), NULL);
+    if (!NT_SUCCESS(queryStatus)) {
+        SetLastError(RtlNtStatusToDosError(queryStatus));
+        return TIME_ZONE_ID_INVALID;
+    }
+
+    lpTimeZoneInformation->Bias = timeZoneBias;
+
+    // Check if DST adjustment is enabled
+    queryStatus = ExQueryNonVolatileSetting(XC_MISC, &type, &dayLightEnFlag, sizeof(dayLightEnFlag), NULL);
+    if (!NT_SUCCESS(queryStatus)) {
+        SetLastError(RtlNtStatusToDosError(queryStatus));
+        return TIME_ZONE_ID_INVALID;
+    }
+
+    // DST adjustment is disabled.
+    if (dayLightEnFlag & 0x2) {
+        return TIME_ZONE_ID_UNKNOWN;
+    }
+
+    // Query all DST info from EEPROM. //FIXME. One large EEPROM access could be better
+    queryStatus = ExQueryNonVolatileSetting(XC_TZ_DLT_DATE, &type, &dayLightDate, sizeof(dayLightDate), NULL);
+    if (!NT_SUCCESS(queryStatus)) {
+        SetLastError(RtlNtStatusToDosError(queryStatus));
+        return TIME_ZONE_ID_INVALID;
+    }
+    queryStatus = ExQueryNonVolatileSetting(XC_TZ_STD_DATE, &type, &standardDate, sizeof(standardDate), NULL);
+    if (!NT_SUCCESS(queryStatus)) {
+        SetLastError(RtlNtStatusToDosError(queryStatus));
+        return TIME_ZONE_ID_INVALID;
+    }
+    queryStatus = ExQueryNonVolatileSetting(XC_TZ_DLT_BIAS, &type, &dayLightBias, sizeof(dayLightBias), NULL);
+    if (!NT_SUCCESS(queryStatus)) {
+        SetLastError(RtlNtStatusToDosError(queryStatus));
+        return TIME_ZONE_ID_INVALID;
+    }
+    queryStatus = ExQueryNonVolatileSetting(XC_TZ_STD_BIAS, &type, &standardBias, sizeof(standardBias), NULL);
+    if (!NT_SUCCESS(queryStatus)) {
+        SetLastError(RtlNtStatusToDosError(queryStatus));
+        return TIME_ZONE_ID_INVALID;
+    }
+    queryStatus = ExQueryNonVolatileSetting(XC_TZ_DLT_NAME, &type, dayLightName, sizeof(dayLightName), NULL);
+    if (!NT_SUCCESS(queryStatus)) {
+        SetLastError(RtlNtStatusToDosError(queryStatus));
+        return TIME_ZONE_ID_INVALID;
+    }
+    queryStatus = ExQueryNonVolatileSetting(XC_TZ_STD_NAME, &type, standardName, sizeof(standardName), NULL);
+    if (!NT_SUCCESS(queryStatus)) {
+        SetLastError(RtlNtStatusToDosError(queryStatus));
+        return TIME_ZONE_ID_INVALID;
+    }
+
+    if (standardDate.Month > 12 || dayLightDate.Month > 12 || 
+        standardDate.DayOfWeek > 6 || dayLightDate.DayOfWeek > 6 ||
+        standardDate.Hour > 23 || dayLightDate.Hour > 23) {
+        return TIME_ZONE_ID_INVALID;
+    }
+
+    // Xbox stores up to 4 characters here.
+    for (UCHAR i = 0; i < 4; i++) {
+        lpTimeZoneInformation->StandardName[i] = standardName[i];
+        lpTimeZoneInformation->DaylightName[i] = dayLightName[i];
+    }
+
+    // There was no Daylight savings date info
```

Comment:
Sometimes it's "Daylight savings", and sometimes it's "day light savings".

Should probably be "daylight savings" ?

Reply:
Made it daylight savings time but now use the abbreviation DST is most places now